.github/workflows/cicd.yml

```
@@ -16,691 +16,51 @@ on:
-  deploy-staging:
-    if: github.ref == 'refs/heads/main'
-    name: (staging) Deploy
+    needs: push
+    if: github.event_name == 'pull_request'
+    uses: ./.github/workflows/deploy-pr.yml
+    secrets: inherit
+
+  deploy:
+    needs: push
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    uses: ./.github/workflows/deploy.yml
```

actionlint (1.7.7)

input "github-sha" is required by "./.github/workflows/deploy.yml" reusable workflow

(workflow-call)